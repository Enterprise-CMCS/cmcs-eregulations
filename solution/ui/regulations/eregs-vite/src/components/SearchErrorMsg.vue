<script setup>
defineProps({
    searchQuery: {
        type: String,
        default: "",
    },
    showApology: {
        type: Boolean,
        default: false,
    },
    surveyUrl: {
        type: String,
        default: "",
    },
});
</script>

<template>
    <div class="error-msg__container">
        <span class="error__msg" data-testid="error__msg">
            <span
                v-if="showApology"
                class="error__msg--apology"
            >Sorry, weâ€™re unable to display results<span
                v-if="searchQuery"
                class="error__msg--query"
            >
                for <strong>{{ searchQuery }}</strong></span>
                right now.
            </span>
            <span class="error-msg__common">Please try a different query, try again later, or
                <a
                    :href="surveyUrl"
                    target="_blank"
                    rel="noopener noreferrer"
                >let us know</a>.</span></span>
    </div>
</template>
