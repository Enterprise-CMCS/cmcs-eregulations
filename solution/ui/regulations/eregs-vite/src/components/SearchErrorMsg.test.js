import flushPromises from "flush-promises";
import { render, screen } from "@testing-library/vue";
import { describe, it, expect } from "vitest";

import SearchErrorMsg from "./SearchErrorMsg.vue";

describe("Search Error Message", () => {
    it("Renders a message with a search query", async () => {
        const wrapper = render(SearchErrorMsg, {
            props: {
                searchQuery: "Search Query",
                showApology: true,
                surveyUrl: "Survey URL",
            },
        });

        await flushPromises();

        const errorTextEl = screen.getByTestId(
            "error__msg"
        );

        expect(errorTextEl.textContent).toBe(
            "Sorry, we’re unable to display results for Search Query right now. Please try a different query, try again later, or let us know."
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("Renders a message without a search query", async () => {
        const wrapper = render(SearchErrorMsg, {
            props: {
                searchQuery: "",
                showApology: true,
                surveyUrl: "Survey URL",
            },
        });

        await flushPromises();

        const errorTextEl = screen.getByTestId(
            "error__msg"
        );

        expect(errorTextEl.textContent).toBe(
            "Sorry, we’re unable to display results right now. Please try a different query, try again later, or let us know."
        );

        expect(wrapper).toMatchSnapshot();
    });

    it("Renders a message without an apology", async () => {
        const wrapper = render(SearchErrorMsg, {
            props: {
                searchQuery: "",
                surveyUrl: "Survey URL",
            },
        });

        await flushPromises();

        const errorTextEl = screen.getByTestId(
            "error__msg"
        );

        expect(errorTextEl.textContent).toBe(
            "Please try a different query, try again later, or let us know."
        );

        expect(wrapper).toMatchSnapshot();
    });
});
