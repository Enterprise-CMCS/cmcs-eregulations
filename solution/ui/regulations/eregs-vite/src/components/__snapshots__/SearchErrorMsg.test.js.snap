// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search Error Message > Renders a message with a search query 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="error-msg__container"
      >
        <span
          class="error__msg"
          data-testid="error__msg"
        >
          <span
            class="error__msg--apology"
          >
            Sorry, we’re unable to display results
            <span
              class="error__msg--query"
            >
               for 
              <strong>
                Search Query
              </strong>
            </span>
             right now. 
          </span>
          <span
            class="error-msg__common"
          >
            Please try a different query, try again later, or 
            <a
              href="Survey URL"
              rel="noopener noreferrer"
              target="_blank"
            >
              let us know
            </a>
            .
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="error-msg__container"
    >
      <span
        class="error__msg"
        data-testid="error__msg"
      >
        <span
          class="error__msg--apology"
        >
          Sorry, we’re unable to display results
          <span
            class="error__msg--query"
          >
             for 
            <strong>
              Search Query
            </strong>
          </span>
           right now. 
        </span>
        <span
          class="error-msg__common"
        >
          Please try a different query, try again later, or 
          <a
            href="Survey URL"
            rel="noopener noreferrer"
            target="_blank"
          >
            let us know
          </a>
          .
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Search Error Message > Renders a message without a search query 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="error-msg__container"
      >
        <span
          class="error__msg"
          data-testid="error__msg"
        >
          <span
            class="error__msg--apology"
          >
            Sorry, we’re unable to display results
            <!--v-if-->
             right now. 
          </span>
          <span
            class="error-msg__common"
          >
            Please try a different query, try again later, or 
            <a
              href="Survey URL"
              rel="noopener noreferrer"
              target="_blank"
            >
              let us know
            </a>
            .
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="error-msg__container"
    >
      <span
        class="error__msg"
        data-testid="error__msg"
      >
        <span
          class="error__msg--apology"
        >
          Sorry, we’re unable to display results
          <!--v-if-->
           right now. 
        </span>
        <span
          class="error-msg__common"
        >
          Please try a different query, try again later, or 
          <a
            href="Survey URL"
            rel="noopener noreferrer"
            target="_blank"
          >
            let us know
          </a>
          .
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Search Error Message > Renders a message without an apology 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="error-msg__container"
      >
        <span
          class="error__msg"
          data-testid="error__msg"
        >
          <!--v-if-->
          <span
            class="error-msg__common"
          >
            Please try a different query, try again later, or 
            <a
              href="Survey URL"
              rel="noopener noreferrer"
              target="_blank"
            >
              let us know
            </a>
            .
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="error-msg__container"
    >
      <span
        class="error__msg"
        data-testid="error__msg"
      >
        <!--v-if-->
        <span
          class="error-msg__common"
        >
          Please try a different query, try again later, or 
          <a
            href="Survey URL"
            rel="noopener noreferrer"
            target="_blank"
          >
            let us know
          </a>
          .
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
